### PR TITLE
babili npm package testing

### PR DESCRIPTION
**Pull request by garyjN7**

### CODE REVIEW

File: scripts/benchmark.js [hunk 1]
```diff
@@ -12,14 +12,28 @@ const babel  = require("babel-core");
 const zlib   = require("zlib");
 const fs     = require("fs");
 const path   = require("path");
+const commander = require("commander").Command;
```
Comment (chrisprice) [inline scripts/benchmark.js]: How come this doesn't look like the commander readme?

Comment (chrisprice) [reply scripts/benchmark.js]: i.e. 

``` js
const program = require('commander');
program.command('setup [env]')
  .description('run setup commands for all envs')
  .option("-s, --setup_mode [mode]", "Which setup mode to use")
  .action(function(env, options){
    var mode = options.setup_mode || "normal";
    env = env || 'all';
    console.log('setup for %s env(s) with %s mode', env, mode);
  });
```

File: scripts/benchmark.js [hunk 2]
```diff
@@ -46,10 +60,79 @@ const table = new Table({
...
+}
+
+function checkFile() {
+  // If filename has not been passed as an argument, attempt to resolve file from package.json
+  filename = filename ? path.join(pathToScripts, "node_modules", filename) : require.resolve(packagename.split("@")[0]);
+  console.log("file: " + path.basename(filename));
+
+  if(!filename || !pathExists(filename)) {
+    console.error("File not found. Exiting.");
+	!script.offline && uninstallPackage();
```
Comment (chrisprice) [inline scripts/benchmark.js]: Indentation looks a bit wobbly here, have you run the lint task?

Comment (chrisprice) [reply scripts/benchmark.js]: I think a standard `if` would be clearer

File: scripts/benchmark.js [hunk 3]
```diff
@@ -46,10 +60,79 @@ const table = new Table({
...
+	process.exit(1);
+  }
+}
+
+function testFile() {
```
Comment (chrisprice) [inline scripts/benchmark.js:129]: Could we move this function below `test` to make the diff line up more cleanly?

File: scripts/benchmark.js [hunk 4]
```diff
@@ -78,62 +161,54 @@ function test(name, callback) {
...
+if(!pathExists(path.join(pathToScripts, "node_modules", packagename)) || !script.offline) {
+  installPackage();
+}
+
+checkFile();
+testFile();
+processResults();
 
-console.log(table.toString());
+!script.offline && uninstallPackage();
```
Comment (chrisprice) [inline scripts/benchmark.js]: Same comment about `if`

File: scripts/benchmark.js [hunk 5]
```diff
@@ -78,62 +161,54 @@ function test(name, callback) {
...
-  if (i === 0) {
-    style = chalk.green;
+function pathExists(path) {
+  try {
+    return fs.statSync(path);
   }
-  if (i === results.length - 1) {
-    style = chalk.red;
+  catch(e) {
+	return false;
```
Comment (chrisprice) [inline scripts/benchmark.js]: Same comment about shonky indentation

File: scripts/benchmark.js [hunk 6]
```diff
@@ -78,62 +161,54 @@ function test(name, callback) {
...
   }
-  row = row.map(function (item) {
-    return style(item);
-  });
+}
 
-  table.push(row);
-});
+if(!pathExists(path.join(pathToScripts, "node_modules", packagename)) || !script.offline) {
```
Comment (chrisprice) [inline scripts/benchmark.js]: This condition is quite hard to follow, is there anyway we can make it simpler?

Comment (chrisprice) [reply scripts/benchmark.js]: Or maybe move it into the `installPackage` function?

File: scripts/benchmark.js (hunk 4 above)
Comment (chrisprice) [reply scripts/benchmark.js]: And maybe move it into the function?

**Review by chrisprice — CHANGES_REQUESTED** [review]
Almost there :)

File: scripts/benchmark.js (hunk 1 above)
Comment (garyjN7) [reply scripts/benchmark.js]: Because this was the only way I could see to fix my problem of the help message i.e. I needed to overwrite the script name and this is the only way I could do it I could see.

File: scripts/benchmark.js (hunk 2 above)
Comment (garyjN7) [reply scripts/benchmark.js]: The indenting looks fine when viewed in the actual script?  I will manually space it to see if that fixes it.

File: scripts/benchmark.js (hunk 1 above)
Comment (chrisprice) [reply scripts/benchmark.js]: It's more normal to use the imported name as the alias e.g. -

``` js
const Command = require("commander").Command;
//...
const script = new Command("benchmark.js")
```

**Review by chrisprice — CHANGES_REQUESTED** [review]
Just the one minor thing. Diff looks much cleaner now - https://github.com/chrisprice/babili/pull/2/files?w=0

File: scripts/benchmark.js (hunk 1 above)
Comment (garyjN7) [reply scripts/benchmark.js]: changed

**Review by chrisprice — APPROVED** [review]
